processor/transformprocessor/README.md

```diff
@@ -51,6 +51,7 @@ Metric only functions:
 Supported where operations:
 - `==` - matches telemetry where the values are equal to each other
 - `!=` - matches telemetry where the values are not equal to each other
+- `like` - matches telemetry where the left string value matches the glob on the right.  If the right is not a string literal the processor will error during startup.  If the left value is not a string literal, is not a path to a string, or is not a function that returns a string, the processor will panic during processing.
```

Comment:
Another option is to use a factory and keep the `==`, `!=`

`name == RegExp("health.*")`

I think we will definitely need regex, anytime you add glob to something someone then asks for regex ;) We wouldn't want the grammar to explode supporting multiple types of matching paradigms though, which this avoids. What do you think?

Reply:
I like how that keeps the operators simpler and I agree that someone will instantly ask for Regex lol

Looking at this syntax tho I'm not sure how we handle the actual check without some complicated smarts in the grammar. Right now the left and right side of the operators are always getters, and the values returned are hard coded to be compared with == or !=. With the Regex Factory we'd have to know the right side is a Regex and change the comparison. That discovery would have to be during runtime bc we can't call functions during initialization.

Reply:
Actually maybe instead of forcing conditions to have an operator they just need to evaluate to a bool 🤔. We could do "... where match_regex(pattern, Getter)" with the implied " == true" (or force it to be there, the grammar wouldn't need changed at all in that case.

Reply:
I guess the issue here is these are only for comparison, and shouldn't be used outside anyways. Perhaps the middle ground is to stick with the grammar change, but instead of `like` (a new operator), introduce syntax for the right side.

```
r"health.*"
g"health*"
```

Or these could even be `RegExp` and `Glob`, they're not actually functions but parsed by the grammar itself. The risk is that they could be confusing in seeming like they should be supported outside of comparisons - but perhaps no one would think of trying this anyways.

Reply:
I would say if we want to enforce something to be usable only in conditions then a new operator is the right tool. Operators don't appear anywhere else, so they are the fullproof way to limit the "check a string against a Regex pattern" Boolean need to conditions only. 

Using r'health.*' as a regex literal I think would work bc while constructing the condition function we can check if the right side is a literal regex like this PR does for string. But then we are still messing with the actual condition check, we would be doing an actual "==" comparison.

Reply:
And if we want to support both glob and regex I think a Factory that returns a book is more flexible. And that Factory could be usable elsewhere. Maybe someone would want to use the Factories as the Getter in the set function)

Reply:
Ah a Factory returning a bool (so I guess `MatchRegexp(name, "health.*")`) sounds like it could be pretty clear